lib/PGresource.pm

```diff
@@ -14,110 +14,137 @@
-	# $self->warning_message( "Test warning message from resource object");
-	# Use this to check if the warning and debug channels have been hooked up to PGcore and PGalias correctly.
-	return $self;
-}
 
-sub uri {
-	my $self = shift;
-	my $uri  = shift;
-	$self->{uri}->{content} = $uri if $uri;
-	$self->{uri}->{content};
-}
+	$self->warning_message("PGresource must be called with a name.") unless $id;
+	$self->warning_message("PGresource must be called with a type.") unless $type;
 
-sub path {
-	my $self = shift;
-	my $url  = shift;
-	$self->{path}->{content} = $url if $url;
-	$self->{path}->{content};
+	# Use this to check if the warning and debug channels have been hooked up to PGcore and PGalias correctly.
+	#$self->warning_message("Test warning message from resource object");
+	#$self->debug_message("Test debug message from resource object");
+
+	return $self;
 }
 
 sub create_unique_id {
-	my $self     = shift;
-	my $ext      = shift;
+	my ($self, $ext) = @_;
 	my $fileName = $self->fileName;
+
 	if ($self->{unique_id}) {
 		$self->warning_message("unique id already exists for $fileName.");
 		return $self->{unique_id};
 	}
-	$self->warning_message("auxiliary file $fileName missing resource path ") unless $self->path;
-	$self->warning_message("auxiliary file $fileName missing problem psvn")   unless $self->{parent_alias}->{psvn};
-	$self->warning_message("auxiliary file $fileName missing unique_id_stub")
-		unless $self->{parent_alias}->{unique_id_stub};
-	my $unique_id_seed = $self->path() . $self->{parent_file_id} . $self->{id};
+
+	$self->warning_message(qq{Auxiliary file "$fileName" missing resource path.}) unless $self->path;
+	$self->warning_message(qq{Auxiliary file "$fileName" missing unique_id_stub.})
+		unless $self->{parent_alias}{unique_id_stub};
+
+	my $unique_id_seed = $self->path . $self->{probFileName} . $self->{id};
 	$self->{unique_id} =
-		$self->{parent_alias}->{unique_id_stub} . '___' . create_uuid_as_string(UUID_V3, UUID_NS_URL, $unique_id_seed);
+		$self->{parent_alias}{unique_id_stub} . '___' . create_uuid_as_string(UUID_V3, UUID_NS_URL, $unique_id_seed);
 	$self->{unique_id} .= ".$ext" if $ext;
-	$self->{unique_id};
 
+	return $self->{unique_id};
+}
+
+sub uri {
+	my ($self, $uri) = @_;
+	$self->{uri} = $uri if $uri;
+	return $self->{uri};
+}
+
+sub path {
+	my ($self, $path) = @_;
+	$self->{path} = $path if $path;
+	return $self->{path};
 }
 
 sub unique_id {
-	my $self      = shift;
-	my $unique_id = shift;
-	$self->{unique_id} = $unique_id if $unique_id;
-	$self->{unique_id};
+	my $self = shift;
+	return $self->{unique_id};
 }
 
 sub fileName {
-	my $self     = shift;
-	my $fileName = shift;
+	my ($self, $fileName) = @_;
 	$self->{id} = $fileName if $fileName;
-	$self->{id};
+	return $self->{id};
 }
+
 1;
+
+=head1 NAME
+
+PGresource - Store information for an auxiliary resource.
+
+=head2 new
+
+Usage: C<< PGalias->new($parent_alias, $id, $type, %options) >>
```

Comment:
Should this be `PGresource->new` instead of `PGalias->new` ?

Reply:
Yeah.  Thanks.